Add procedures to migrate from RHEL8to9 using backups (#3103)
theforeman/foreman-documentation: guides/common/modules/proc_migrating_project_or_smartproxy_using_backup.adoc

[id="migrating-{project-context}-or-proxy-using_backup_{context}"]
= Migrating {Project} or {SmartProxy} to {EL}{nbsp}9

You can migrate your existing {Project} and {SmartProxy} installation from {EL}{nbsp}8 to a freshly installed {EL}{nbsp}9 system.
ifdef::satellite[]
The migration process involves creating a backup of the existing {Project} and {SmartProxy} installation, which is then restored or cloned on the new {EL}{nbsp}9 system.

> Note that cloning only works for {Project} backups and not for {SmartProxy} backups.

Reviewer: +1.

I recommend using the note admonition format instead of "Note that ...".

endif::[]
ifndef::satellite[]
The migration process involves creating a backup of the existing {Project} and {SmartProxy} installation, which is then restored on the new {EL}{nbsp}9 system.

> endif::[]

Author: So the statement above mentions the `restoring and cloning` which are only available for Satellite, while this is for non-Satellite builds as they do not have the option to clone.

In the below procedure, the "Source server" refers to the existing {EL}{nbsp}8 system on which {Project} or {SmartProxy} is installed, while "Target server" refers to the new {EL}{nbsp}9 system to which {Project} or {SmartProxy} will be migrated.

.Procedure:
. Ensure that you have enough space for the backup.

> For more information, see {AdministeringDocURL}administering_red_hat_satellite/backing-up-satellite-server-and-capsule_admin#Estimating_the_Size_of_a_Backup_admin[Estimating the size of a backup] in _{AdministeringDocTitle}_.

Author: Ack. I think removing this makes more sense, since the prerequisites already mention it.

. Perform a full backup of the {Project} or {SmartProxy}.
For more information, see {AdministeringDocURL}backing-up-satellite-server-and-capsule_admin#Performing_a_Full_Backup_admin[Performing a full backup of Satellite Server or Capsule Server] in _{AdministeringDocTitle}_.

> . Deploy a new {EL}{nbsp}9 system.

Reviewer: ```suggestion
. Deploy a {EL}{nbsp}9 system.
```
If the user is deploying a system, then it is 'new' by definition.

> Choose the appropriate method to install {Project} or {SmartProxy}:

Reviewer: I don't think you need "Choose ..." The user can see what the choices are.

> Choose the appropriate method to install {Project} or {SmartProxy}:

Reviewer: ```suggestion
. Install {Project} or {SmartProxy} on the {EL}{nbsp}9 system:
```
This looks like a separate step to me.

> Choose the appropriate method to install {Project} or {SmartProxy}:

Author: Makes sense to me, I was a little iffy about the definitions in the intro, since they can be easily missed and the usage of "in the below procedure" made it sound somewhat clunky too. :)

** To install {ProjectServer} from a connected network, follow the procedures in {InstallingServerDocURL}[{InstallingServerDocTitle}].
ifdef::satellite[]
** To install {ProjectServer} from a disconnected network, follow the procedures in {InstallingServerDisconnectedDocURL}[{InstallingServerDisconnectedDocTitle}].
endif::[]
** To install a {SmartProxyServer}, follow the procedures in {InstallingSmartProxyDocURL}[{InstallingSmartProxyDocTitle}].
ifdef::satellite[]

> . Use either of these two procedures to restore backup from {Project}: 

Reviewer: If we offer more than one method to perform a task, we need to describe the use cases.

Explain the difference between the two methods and why a user would want to use one or the other.

Author: ack @apinnick. Let me check more about the pros and cons of these two procedures.

> . Use either of these two procedures to restore backup from {Project}: 

Author: Satellite can be restored with either restore a clone, but capsules can only be restored. As to their pros and cons, I'll check with the SME and add it to the intro.

** {AdministeringDocURL}restoring-satellite-server-or-capsule-from-a-backup_admin[Restoring Satellite Server or Capsule Server from a backup] in _{AdministeringDocTitle}_.
** {AdministeringDocURL}cloning_satellite_server#sec-Cloning_to_Target[Cloning to the target server] in _{AdministeringDocTitle}_.

> . To restore the {SmartProxy} backup, follow the steps in {AdministeringDocURL}restoring-satellite-server-or-capsule-from-a-backup_admin[Restoring Satellite Server or Capsule Server from a backup] in _{AdministeringDocTitle}_. 

Reviewer 2: I advise against using "procedure". It's [self-referential](https://redhat-documentation.github.io/supplementary-style-guide/#shortdesc). Furthermore, the trouble with the word "procedure" specifically is that users might actually not know what a "procedure" is (is it the whole section? or is it just whatever is under `.Procedure` without any prerequisites?) For there reasons, I prefer our standard wording based on "For more information".

> . To restore the {SmartProxy} backup, follow the steps in {AdministeringDocURL}restoring-satellite-server-or-capsule-from-a-backup_admin[Restoring Satellite Server or Capsule Server from a backup] in _{AdministeringDocTitle}_. 

Author: Because only Satellite backups can be either cloned or restored on the target, while Capsule backups only have the option to restore, I created two separate statements. But yes, they do refer to the same links.

endif::[]
ifndef::satellite[]

> . Restore the backups from {Project} or {SmartProxy}. For more information, see {AdministeringDocURL}restoring-satellite-server-or-capsule-from-a-backup_admin[Restoring Satellite Server or Capsule Server from a backup] in _{AdministeringDocTitle}_.

Reviewer 2: The links here and elsewhere include hard-coded "satellite". I think that's causing issues for non-Satellite builds. Can you please check?

[id="migrating-{project-context}-or-proxy-using_backup_{context}"]
= Migrating {Project} or {SmartProxy} to {EL}{nbsp}9

> You can migrate your existing {Project} and {SmartProxy} installation from {EL}{nbsp}8 to a freshly installed {EL}{nbsp}9 system.

Reviewer 3: ```suggestion
You can migrate your existing {ProjectServer} and {SmartProxyServer} from {EL}{nbsp}8 to a freshly installed {EL}{nbsp}9 system.
```

ifdef::satellite[]

> The migration involves creating a backup of the existing {Project} and {SmartProxy} installation, which you then restore or clone on the new {EL}{nbsp}9 system.

Reviewer 3: ```suggestion
The migration involves creating a backup of the existing {ProjectServer} and {SmartProxyServers}, which you then restore or clone on the new {EL}{nbsp}9 system.
```

[NOTE]
====

> Cloning only works for {Project} backups and not for {SmartProxy} backups.

Reviewer 3: ```suggestion
Cloning only works for {ProjectServer} backups and not for {SmartProxyServer} backups.
```

====
endif::[]
ifndef::satellite[]

> The migration involves creating a backup of the existing {Project} and {SmartProxy} installation, which you then restore on the new {EL}{nbsp}9 system.

Reviewer 3: ```suggestion
The migration involves creating a backup of the existing {ProjectServer} and {SmartProxyServers}, which you then restore on the new {EL}{nbsp}9 system.
```

endif::[]

.Procedure

> . Perform a full backup of the {Project} or {SmartProxy}.

Reviewer 3: ```suggestion
. Perform a full backup of your {ProjectServer} or {SmartProxyServer}.
```

> This is the source {EL}{nbsp}8server that you are migrating.

Reviewer 3: ```suggestion
This is the source {EL}{nbsp}8 server that you are migrating.
```

> For more information, see {AdministeringDocURL}backing-up-satellite-server-and-capsule_admin#Performing_a_Full_Backup_admin[Performing a full backup of Satellite Server or Capsule Server] in _{AdministeringDocTitle}_.

Reviewer 3: ```suggestion
For more information, see {AdministeringDocURL}backing-up-{project-context}-server-and-{smart-proxy-context}_admin#Performing_a_Full_Backup_admin[Performing a full backup of {ProjectServer} or {SmartProxyServer}] in _{AdministeringDocTitle}_.
```

. Deploy a {EL}{nbsp}9 system.
This is the target server.

> Install {Project} or {SmartProxy} on the target {EL}{nbsp}9 system:

Reviewer 3: ```suggestion
Install {ProjectServer} or {SmartProxyServer} on the target {EL}{nbsp}9 system:
```

** To install {ProjectServer} from a connected network, see {InstallingServerDocURL}[{InstallingServerDocTitle}].
ifdef::satellite[]
** To install {ProjectServer} from a disconnected network, see {InstallingServerDisconnectedDocURL}[{InstallingServerDisconnectedDocTitle}].
endif::[]
** To install a {SmartProxyServer}, see {InstallingSmartProxyDocURL}[{InstallingSmartProxyDocTitle}].
ifdef::satellite[]
. Use either of these two procedures to restore backup from {Project}: 

> ** {AdministeringDocURL}restoring-satellite-server-or-capsule-from-a-backup_admin[Restoring Satellite Server or Capsule Server from a backup] in _{AdministeringDocTitle}_.

Reviewer 3: ```suggestion
** {AdministeringDocURL}restoring-{project-context}-server-or-{smart-proxy-context}-from-a-backup_admin[Restoring {ProjectServer} or {SmartProxyServer} from a backup] in _{AdministeringDocTitle}_.
```

> ** {AdministeringDocURL}cloning_satellite_server#sec-Cloning_to_Target[Cloning to the target server] in _{AdministeringDocTitle}_.

Reviewer 3: ```suggestion
** {AdministeringDocURL}cloning_{project-context}_server#sec-Cloning_to_Target[Cloning to the target server] in _{AdministeringDocTitle}_.
```

. To restore the {SmartProxy} backup, follow the steps in {AdministeringDocURL}restoring-satellite-server-or-capsule-from-a-backup_admin[Restoring Satellite Server or Capsule Server from a backup] in _{AdministeringDocTitle}_. 
endif::[]
ifndef::satellite[]

> . Restore the backups from {Project} or {SmartProxy}. For more information, see {AdministeringDocURL}restoring-foreman-server-or-smart-proxy-from-a-backup_admin[Restoring Foreman server or Smart Proxy server from a backup] in _{AdministeringDocTitle}_.

Reviewer 3: ```suggestion
. Restore the backups from {Project} or {SmartProxy}.
For more information, see {AdministeringDocURL}restoring-{project-context}-server-or-{smart-proxy-context}-from-a-backup_admin[Restoring {ProjectServer} or {SmartProxyServer} from a backup] in _{AdministeringDocTitle}_.
```

[id="migrating-{project-context}-or-proxy-using_backup_{context}"]
= Migrating {Project} or {SmartProxy} to {EL}{nbsp}9

You can migrate your existing {Project} and {SmartProxy} installation from {EL}{nbsp}8 to a freshly installed {EL}{nbsp}9 system.

.Terminology
Ensure that you understand the following terms:

*Source server*:: The existing {EL}{nbsp}8 system on which {Project} or {SmartProxy} is installed.

*Target server*:: The new {EL}{nbsp}9 system to which {Project} or {SmartProxy} will be migrated.

> 

Reviewer 2: I don't think you need this.

You're not using "source server" anywhere in the procedure => no need to define it.

Rather than define "target server", you could try using `{EL}{nbsp}9` in the procedure. In the context of this procedure, that would leave no room for doubt about what system you're describing.

In any case, I believe using a `.Terminology` subheading is an unnecessary divergence from the module template.

Author: @asteflova the procedures to which I am redirecting refer to source and target, the definition for these in the backups chapter introduction differs from the context in which they should be used here. But I do agree that the divergence should be avoided. Should I add these definition sans the `.Terminology` in the introduction?

.Migration process overview:
. Create a backup of the existing {Project} or {SmartProxy} on {EL}{nbsp}8.
. Deploy a new {EL}{nbsp}9 system.
ifdef::satellite[]
. Restore the backup to the target server using one of these two methods:
.. Restoring backup: You can restore both {Project} and {SmartProxy} backups with this method.
.. Clone: You can only restore {Project} backup with this method.
endif::[]
ifndef::satellite[]
. Restore the backup to the target server.
endif::[]

> 

Reviewer 2: No need to diverge from the template with a subheading. This is actually what I'm missing in the procedure intro on line 4. Can you consider rewriting this to fit the introduction? I'm thinking of something like "During migration, you first create a backup [...]". If you do this, I'm sure you can also drop some of the details. Users can get an overview of the migration by looking at the procedure itself, lines 13-23 basically duplicate the steps of the procedure.
